src/current/v25.2/architecture/transaction-layer.md

```diff
@@ -432,6 +432,38 @@ Despite their logical equivalence, the transaction coordinator now works as quic
+
+Buffered Writes update the transaction flow so that it has the following properties:
+
+1. Write buffering: Instead of sending each write operation immediately to the [leaseholder]({% link {{ page.version.version }}/architecture/overview.md %}#architecture-leaseholder), writes are now buffered on the gateway node until the transaction commits. The writes can then be issued in a single batch, reducing the number of total requests required to complete the transaction.
+1. Local read-your-writes: During the transaction, any read operation that targets a key with a buffered write can be served directly from the buffer, avoiding [pipeline](#transaction-pipelining) stalls and reducing latency.
+1. Write batching: At commit time, buffered writes are batched and sent to the [storage layer]({% link {{ page.version.version }}/architecture/storage-layer.md %}), reducing the frequency of write operations and improving throughput.
+1. 1-phase commit optimization (1PC): If all writes in the buffer target the same [range]({% link {{ page.version.version }}/architecture/overview.md %}#architecture-range), the system can utilize the [1-phase commit fast path]({% link {{ page.version.version }}/ui-distributed-dashboard.md %}#kv-transactions), further optimizing the commit process. Explicit transactions that formerly were not eligible for 1PC may become eligible because their writes have been buffered.
+1. Redundant write elimination: Within a single transaction, in many cases only the final write operation for each key is sent to the [Storage layer]({% link {{ page.version.version }}/architecture/storage-layer.md %}) at commit time, eliminating redundant operations.
+
+Buffered Writes are off by default. They can be enabled using either a [session setting]({% link {{ page.version.version }}/session-variables.md %}) or a [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}).
+
+To turn them on for the current session, issue the following statement:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET kv.transaction.buffered_writes.enabled = true
+~~~
+
+{{site.data.alerts.callout_info}}
+Buffered Writes has the following limitations:
+
+- Users of [`READ COMMITTED`]({% link {{ page.version.version }}/read-committed.md %}) isolation won't see a performance benefit. This limitation is likely to remain in place until the feature is [Generally Available (GA)]({% link {{ page.version.version }}/cockroachdb-feature-availability.md %}#feature-availability-phases).
+- Some workloads could see an increase in [transaction retries]({% link {{ page.version.version }}/transactions.md %}#transaction-retries).
```

Comment:
Is it possible to indicate whether these are automatic or client-side or both?

Reply:
updated to clarify they are transaction retry errors (not auto-handled AFAICT), and linked to the txn retry error page 

@stevendanna pls correct me if wrong, and this does include auto-retries - my sense was no!

Reply:
I think it includes both.